qdev_wrappers/fitting/Fitter.py

```diff
@@ -0,0 +1,289 @@
+import qcodes as qc
+import numpy as np
+import dill
+
+from qdev_wrappers.fitting.Converter import SQL_Converter, Legacy_Converter
+from qdev_wrappers.fitting.Fitclasses import T1, T2
+from scipy.optimize import curve_fit
+
+
+def fit_data(data, fitclass, fun_inputs, fun_output, setpoint_params=None, p0=None, **kwargs):
+
+    fun_dim = len(fitclass.fun_vars)
+    num_inputs = len(fun_inputs)
+    if setpoint_params is not None:
+        num_setpoints = len(setpoint_params)
+    else: num_setpoints = 0
+
+
+    if fun_dim == 1 and num_inputs == 1 and num_setpoints == 0:
+        fitter = Fitter_1D(data, fitclass, fun_output)
+    elif fun_dim == 1 and num_inputs == 1 and num_setpoints == 1:
+        fitter.Fitter_2Ddata_1Dfunction(data, fitclass, fun_output)    #what does the fitter actually need??
+    else:
+        print('The specified function has ')
+
+    fitter.find_fit(data, fitclass, fun_inputs, fun_output, setpoint_params, p0, **kwargs)
+
+class Fitter1D:
```

Comment:
maybe I need to read this a bit more carefully but I'm wondering if this is really a Fitter1D or really just a class describing the fit which could have more than 1D? Is this just a Fitter? Essentially I'm asking what the 1D refers to? Also it seems a bit confusing to have both a Fitclass and a Fitter class, this may well be the best way but if so then it is definitely a case where a few lines of description on each and a good paragraph in the example notebook would go a long way :)

Reply:
What makes this specific to 1D is the 'perform_fit' function. I'm not sure how to write that part generically, since it seems to me that the data has to be organized differently depending on the dimensionality of the data and the function. I've changed it to have a general Fitter class as a base, and then two classes that inherit from that, to make it a bit clearer with the function specific to a '1D' fit (i.e. '1D fit to 1D data') or a '1D fit to 2D data'. 

I've gone back and forth a fair bit between it being a class and just being a function. Right now I've landed on a class, because it allows class inheritance if it turns out that there isn't a good way to make perform_fit generic in terms of number of variables. I also couldn't get dill to work for a function, whereas it worked without issue for an instance of a class. I'm not sure what I was doing wrong there.